src/extensions/extensions-api-client-internal.ts

```diff
@@ -14,26 +14,150 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 import { App } from '../app';
 import { FirebaseApp } from '../app/firebase-app';
-import { HttpClient, AuthorizedHttpClient } from '../utils/api-request';
-import { FirebaseAppError } from '../utils/error';
+import { AuthorizedHttpClient, HttpClient, HttpError, HttpRequestConfig } from '../utils/api-request';
+import { FirebaseAppError, PrefixedFirebaseError } from '../utils/error';
 import * as validator from '../utils/validator';
+import * as utils from '../utils';
+
+const FIREBASE_FUNCTIONS_CONFIG_HEADERS = {
+  'X-Firebase-Client': `fire-admin-node/${utils.getSdkVersion()}`
+};
+const EXTENSIONS_API_VERSION = 'v1beta';
+
 /**
  * Class that facilitates sending requests to the Firebase Extensions backend API.
  *
  * @internal
  */
 export class ExtensionsApiClient {
-  /* eslint-disable-next-line no-unused-vars */
   private readonly httpClient: HttpClient;
 
-  constructor(private readonly app: App) {
+  constructor(app: App) {
     if (!validator.isNonNullObject(app) || !('options' in app)) {
       throw new FirebaseAppError(
         'invalid-argument',
         'First argument passed to getExtensions() must be a valid Firebase app instance.');
     }
     this.httpClient = new AuthorizedHttpClient(app as FirebaseApp);
   }
+
+  async getRuntimeData(projectId: string, instanceId: string): Promise<RuntimeDataResponse> {
```

Comment:
We should not include getRuntimeData at this time. We don't think it will be very useful, the BE has no implementation, nor is there planned to be one until/unless people ask for it specifically.